R/formatter.r

```diff
@@ -232,32 +232,70 @@ scientific <- function(x, digits = 3, ...) {
 
 #' Ordinal formatter: add ordinal suffixes (-st, -nd, -rd, -th) to numbers.
 #'
-#' @return a function with single parameter x, a numeric vector, that
+#' @return A function with single parameter `x`, a numeric vector, that
 #'   returns a character vector
-#' @param x a numeric vector to format
+#' @param x A numeric vector of positive values to format.
+#' @param prefix,suffix Symbols to display before and after value.
+#' @param big.mark Character used between every 3 digits to separate thousands.
+#' @param rules Custom rules for computing ordinal indicators (list of regular
+#'   expressions, see examples).
+#' @param ... Other arguments passed on to [base::format()].
+#' @note
+#' Values in `x` will be rounded before formating.
 #' @export
 #' @examples
 #' ordinal_format()(1:10)
 #' ordinal(1:10)
-ordinal_format <- function() {
-  function(x) ordinal(x)
+#'
+#'  # Custom rules for French
+#' french <- list(
+#'   er = "^1$",
+#'   nd = "^2$",
+#'   e = "([0-9]+)[12]$",
```

Comment:
What happens if a number matches multiple regular expressions? If it return the first match, we could make the fallbacks (in both French and English) simpler by matching anything (i.e. `e = '.'`)

Reply:
I have done a quick test. If a number match two or more rules, two or more values will be returned. I will have to update the code to consider only the first results

Reply:
corrected